src/main/ruby/core/truffle/cext.rb

```diff
@@ -129,6 +129,12 @@ def []=(index, value)
     alias_method :to_s, :string
   end
 
+  class RStringPtrEnd < RStringPtr
```

Comment:
This will behave badly if say READ(ptr, 0) is sent. I think it would be better to define just what is needed and consider that this points to the `\0` (so size is 0, etc).